### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Buyaki01**
- Worked on game_logic branch
- Added real logic to the tic tac toe game
- Created three more files: board.rb, players.rb and logic.rb in the lib folder
- board.rb is responsible for the display of the board
- players.rb is responsible for the display of the player's names
- logic.rb is responsible for the logic of the game i.e how to determine the winner and when it's a draw

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,71 @@
...
+require_relative '../lib/players.rb'
+require_relative '../lib/logic.rb'
+
+@board = Board.new
+@winner = ''
+@turn = 'X'
+
+def take_turn(player, cell)
+  @board.board[cell] = player
+end
```
Comment (maelfosso) [inline bin/main.rb]: Move it in `Board`

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+    else
+      puts "#{player2.name} select your cell"
+    end
+    cell = gets.chomp
+
+    while @board.board[cell.to_i] != ' '
+      puts 'Please only select an empty cell between 0 and 8'
+      puts 'Select another cell'
+      cell = gets.chomp
+    end
```
Comment (maelfosso) [inline bin/main.rb]: Check the validity, move into `Board`

File: bin/main.rb [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+  player2 = Player.new(temp_player_name)
+  puts "Player2's name is #{player2.name}"
+
+  while @winner == ''
+    @board.display_board
+    if @turn == 'X'
+      puts "#{player1.name} select your cell"
+    else
+      puts "#{player2.name} select your cell"
+    end
```
Comment (maelfosso) [inline bin/main.rb]: Switching the current player or indicate who is the current player, should come from `Board`

File: bin/main.rb [hunk 4]
```diff
@@ -1,2 +1,71 @@
...
+
+    @winner = @turn if check_win(@board.board) == true
+
+    @winner = 'T' if @winner == '' && !@board.board.include?(' ')
+
+    @turn = if @turn == 'X'
+              'O'
+            else
+              'X'
+            end
```
Comment (maelfosso) [inline bin/main.rb]: Improve your writing code.
It's similar to switching the player, move into `Board` class

File: bin/main.rb [hunk 5]
```diff
@@ -1,2 +1,71 @@
...
+
+    while @board.board[cell.to_i] != ' '
+      puts 'Please only select an empty cell between 0 and 8'
+      puts 'Select another cell'
+      cell = gets.chomp
+    end
+
+    take_turn(@turn, cell.to_i)
+
+    @winner = @turn if check_win(@board.board) == true
```
Comment (maelfosso) [inline bin/main.rb]: Check the win should be done into `Board`

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @Buyaki01 @DennisOsido 

Nice introduction to this milestone.
There are some bugs to fix.

1. Error when starting the application
```ruby
$ ./bin/main.rb 
./bin/main.rb: line 6: require_relative: command not found
./bin/main.rb: line 7: require_relative: command not found
./bin/main.rb: line 8: require_relative: command not found
./bin/main.rb: line 10: @board: command not found
./bin/main.rb: line 11: @winner: command not found
./bin/main.rb: line 12: @turn: command not found
./bin/main.rb: line 14: syntax error near unexpected token `('
./bin/main.rb: line 14: `def take_turn(player, cell)'
```
But this is working
```ruby
$ ruby ./bin/main.rb 
0 |1 |2 
  |  |  
__|__|__
3 |4 |5 
  |  |  
__|__|__
6 |7 |8 
  |  |  
  |  | 
Player1 write your name
```

2. Take a look at this screenshot
Screenshot from 2020-05-11 01-49-23
It's possible to enter **Empty name**. Please, I think you should assign a random name is such a case

Except for those issues, everything is OK. The game works well.

3. Your OOP implementation is not good except for the `Player` class.
You need to move some methods from the `bin/main.rb` file. Please, correctly remember the OOP principles before moving any function.

Kindly make the changes and submit a new code review request.
Thanks 
Mael FOSSO

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### Your branch has some merge conflict that you need to solve before requesting for another code review
merge

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Hi Team,
## Project looks good. Congratulation :+1: Approved.
Go ahead and merge to development.
_Goodluck in your next project_ *Happy coding*

**Pull request merged**